### ISSUE TITLE
setArticleTypePre not called for actions

### ISSUE THREAD

**Issue by NETertainer** [zammad/zammad#4528]
### Used Zammad Version

develop and 5.3

### Environment

Only related to source code.

### Actual behaviour

Despite being used in actions setArticleTypePre is not called while setArticleTypePost is.

setArticleTypePost:
https://github.com/zammad/zammad/blob/973be3998bbf4f6957e8ad18301a425e78f11e70/app/assets/javascripts/app/controllers/ticket_zoom/article_new.coffee#L466-L469

setArticleTypePre:
https://github.com/zammad/zammad/blob/973be3998bbf4f6957e8ad18301a425e78f11e70/app/assets/javascripts/app/controllers/ticket_zoom/article_new.coffee#L391-L461

### Expected behaviour

Call setArticleTypePre for actions like setArticleTypePost.
```
for localConfig in @actions()
      if localConfig && localConfig.setArticleTypePre
        localConfig.setArticleTypePre(@type, @ticket, @, signaturePosition)
```

### Steps to reproduce the behaviour

Change article type from note to email and back.
Signature ist not being removed.
https://github.com/zammad/zammad/blob/973be3998bbf4f6957e8ad18301a425e78f11e70/app/assets/javascripts/app/controllers/ticket_zoom/article_action/email_reply.coffee#L321-L326

### I'm sure this is a bug and no feature request or a general question.

yes

**Comment by MrGeneration** [comment]
There's no such branch / version as `master`.-

**Comment by dominikklein** [comment]
Hi @NETertainer 

thanks for the issue, it looks like a duplicate for: https://github.com/zammad/zammad/issues/4453

**Comment by NETertainer** [comment]
@dominikklein Both issues relate to the same symptom. Just wanted to make sure you noticed that I included the offending lines of code and one possible fix.
I stumbled upon this when I was trying to use 'setArticleTypePre' like the email_reply controller does, but could work my way around it with just doing the same in 'setArticleTypePost'. So in reality 'setArticleTypePre' might not even be needed and the real fix could be to change the behavior of the email_reply controller.
